pynitrokey/cli/update.py

```diff
@@ -7,45 +7,96 @@
 # http://opensource.org/licenses/MIT>, at your option. This file may not be
 # copied, modified, or distributed except according to those terms.
 
-import base64
-import hashlib
-import json
+import os
+import click
+import requests
 import sys
 import tempfile
+import json
 import time
 
-import click
-import requests
 import pynitrokey
 from fido2.ctap import CtapError
 from fido2.ctap1 import ApduError
-from pynitrokey import helpers
 
 
 @click.command()
-@click.option("-s", "--serial", help="Serial number of Nitrokey key to target")
-@click.option(
-    "-y", "--yes", is_flag=True, help="Don't ask for confirmation before flashing"
-)
-@click.option(
-    "-lfs",
-    "--local-firmware-server",
-    is_flag=True,
-    default=False,
-    hidden=True,
-    help="Development option: pull firmware from http://localhost:8000",
-)
-@click.option(
-    "--alpha",
-    is_flag=True,
-    default=False,
-    hidden=True,
-    help="Development option: use release refered to by ALPHA_VERSION",
-)
-def update(serial, yes, local_firmware_server, alpha):
+@click.option("-s", "--serial", help="Serial number of Nitrokey key to target",
+              default=None)
+def update(serial):
     """Update Nitrokey key to latest firmware version."""
 
-    update_url = 'https://update.nitrokey.com/'
-    print('Please use {} to run the firmware update'.format(update_url))
-    return
+    #update_url = 'https://update.nitrokey.com/'
+    #print('Please use {} to run the firmware update'.format(update_url))
+    #return
+
+    # Determine target key
+    try:
+        client = pynitrokey.client.find(serial)
+
+    except pynitrokey.exceptions.NoSoloFoundError:
+        print()
+        print("No Nitrokey key found!")
+        print()
+        print("If you are on Linux, are your udev rules up to date?")
+        print("For more, see https://www.nitrokey.com/documentation/installation#os:linux")
+        print()
+        sys.exit(1)
+    except pynitrokey.exceptions.NonUniqueDeviceError:
+        print()
+        print("Multiple Nitrokey keys are plugged in! Please:")
+        print("  * unplug all but one key")
+        print()
+        sys.exit(1)
+    except Exception:
+        print()
+        print("Unhandled error connecting to key.")
+        print("Please report via https://github.com/Nitrokey/pynitrokey/issues/")
+        print()
+        sys.exit(1)
```

Comment:
I would like to have this exception in a log, as it is done in the Nitrokey Start update.